src/components/theme/SearchWidget/SearchWidget.jsx

```diff
@@ -69,30 +67,16 @@ class SearchWidget extends Component {
     });
   }
 
-  /**
-   * On change section
-   * @method onChangeSection
-   * @param {object} event Event object.
-   * @param {bool} checked Section checked.
-   * @returns {undefined}
-   */
-  onChangeSection(event, { checked }) {
-    this.setState({
-      section: checked,
-    });
-  }
-
   /**
    * Submit handler
    * @method onSubmit
    * @param {event} event Event object.
    * @returns {undefined}
    */
   onSubmit(event) {
-    const section = this.state.section ? `&path=${this.props.pathname}` : '';
-    this.props.history.push(
-      `/search?SearchableText=${this.state.text}${section}`,
-    );
+    const path =
+      this.props.pathname?.length > 0 ? `&path=${this.props.pathname}` : '';
```

Comment:
```suggestion
      this.props.pathname?.length > 0 ? `&path=${encodeURIComponent(this.props.pathname)}` : '';
```

Reply:
I like this API: https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams

Reply:
Yes, that one's very nice. In a situation as simple as this one, though, I wouldn't go with URLSearchParams because of the worse browser compatibility it offers.

Reply:
Works fine for our stack: https://github.com/plone/volto/blob/master/package.json#L188